src/pipeline.py

```diff
@@ -23,6 +23,44 @@ def drop(labels=None, axis=0, index=None, columns=None):
     return lambda d1: DplyFrame(d1.pandas_df.drop(labels=labels, axis=axis, index=index, columns=columns))
 
 
+def write_file(file_path, sep=',', index=True):
+    """
+    Write DplyFrame to file.
+
+    Write the DplyFrame to the following file types depending on the file path given: .csv, .xlsx, .json and .pkl.
+
+    :param file_path: the path of the file with proper suffix
+    :param sep: the separator for csv files. Default to be comma
+    :param index: Write the row name by the index of the DplyFrame. Default to be true.
+    """
+    def to_csv(d1):
+        d1.pandas_df.to_csv(file_path, sep=sep, index=index)
+        return d1
+
+    def to_excel(d1):
+        d1.pandas_df.to_excel(file_path, index=index)
+        return d1
+
+    def to_json(d1):
+        d1.pandas_df.to_json(file_path)
+        return d1
+    
+    def to_pickle(d1):
+        d1.pandas_df.to_pickle(file_path)
+        return d1
+
+    if file_path.endswith('.csv'):
+        return to_csv
+    elif file_path.endswith('.xlsx'):
+        return to_excel
+    elif file_path.endswith('.json'):
+        return to_json
+    elif file_path.endswith('.pkl'):
+        return to_pickle
+    else:
+        print("Unfortunately we do not support this file type. Please write the files with the following types: \n")
```

Comment:
I think we should either the dataframe here (I mean, return a function that returns the dataframe), or throw an exception. Otherwise, this could cause an exception downstream if write_file isn't terminating the pipe. Better to be explicit about the error and halt the pipeline or pass through. What do you think?